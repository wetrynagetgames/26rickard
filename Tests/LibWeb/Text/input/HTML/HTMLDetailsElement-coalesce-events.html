<!DOCTYPE html>
<script src="../include.js"></script>
<details id="details9" open>
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
</details>
<script>
    asyncTest(done => {
        details9 = document.getElementById('details9');

        details9.ontoggle = function(e) {
            println(`target.open is ${e.target.open}`)
            if (e.target.open) {
                println("this should not be called")
            } else {
                println("this should be called")
                done()
            }
        }
        details9.open = true;
        println("details9.open set to true")
        println(`details9.open is still ${details9.open}`)

        // End the test by retrigging the event
        details9.open = false;
    });
</script>
