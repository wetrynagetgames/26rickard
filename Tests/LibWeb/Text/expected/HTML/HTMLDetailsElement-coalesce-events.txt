details9.open set to true
details9.open is still true
target.open is false
this should be called
